pkg/kubectl/cmd/util/factory.go

```diff
@@ -176,7 +177,25 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			if cfg.GroupVersion != nil {
 				cmdApiVersion = *cfg.GroupVersion
 			}
-
+			client, err := clients.ClientForVersion(&unversioned.GroupVersion{Version: "v1"})
```

Comment:
I'm pretty sure that this chunk of code will cause something like the `kubectl convert --local` to fail because this will try to make a remote connection and the `ObjectTyper` is probably used.  If this didn't fail tests, it would suggest that branch isn't being checked, which is probably worthy of a separate issue.